packages/jest-util/src/isPromise.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// see ES2015 spec 25.4.4.5, https://stackoverflow.com/a/38339199
+const isPromise = (candidate: unknown): candidate is Promise<unknown> =>
```

Comment:
do we just want to support promises, or `thenables`?

Reply:
I think the mistake people commonly make is using an async function.
Other forms of returning a promise seem like a hard mistake to make already:
```js
describe('thing', () => {
  return connectToDb().then(() => {
    it('does stuff', () => { })
  });
})
```

Returning other thenables seems to be on the same level as returning Observables or something. IMO enough to forbid those like any value with a generic warning in Jest 25.

Reply:
happy to special case promises in an error message for return values

Reply:
does this work with untranspiled `async/await` as well?